### PR TITLE
Soret coefficient as a function of temperature

### PR DESCRIPTION
**Pull request by KulaginVladimir**
This PR makes it possible to set the Soret coefficient as a function of temperature, according to #458. Following the discussion, the parameter is renamed to `Q` and expressed in eV. Most of the changes are made based on #468.

Basic usage:
```python
from FESTIM import Material

my_mat = Material(id=1, D_0=1, E_D=0, Q=lambda T: 2 + 5 * T)
# or
my_mat = Material(id=1, D_0=1, E_D=0, Q=10)

``` 
## Types of changes

What types of changes does your code introduce to FESTIM?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code refactoring
- [x] Documentation Update (if none of the other choices apply)
- [ ] New tests

## Checklist

- [x] Black formatted
- [x] Unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by jhdark** [comment]
@mougenj Its finally coming!

**Review by RemDelaporteMathurin — CHANGES_REQUESTED** [review]
Thanks for this PR @KulaginVladimir , just a small comment on the docstrings, we could add an example of using a `lambda` function.

Also, do we have an MMS test for the soret effect? I don't know if we tested it thoroughly or not. Can be done in a separate PR if we don't

File: festim/materials/material.py [hunk 1]
```diff
@@ -18,9 +18,8 @@ class Material:
             be a function of T. Defaults to None.
         borders (list, optional): The borders of the 1D subdomain.
             Only needed in 1D with several materials. Defaults to None.
-        H (dict, optional): heat of transport (J/mol).
-            {"free_enthalpy": ..., "entropy": ...} so that
-            H = free_enthalpy + entropy*T. Defaults to None.
+        Q (float or callable, optional): heat of transport (eV). Can
```
[the remark this answers is not in the document]
Comment (KulaginVladimir) [reply festim/materials/material.py:21]: Well, `thermal_cond`, `heat_capacity`, `rho` also can be temperature-dependent. Thus, wouldn't it be beneficial to provide a more general example via [`Example ::`](https://github.com/festim-dev/FESTIM/blob/2ae06c1357194c105d6e4d415c88d105a11258fd/festim/sources/source_implantation_flux.py#L22-L29), e.g.:
```python
my_mat = Material(
    id = 1,
    D_0 = 2e-7,
    E_d = 0.2,
    thermal_cond = lambda T: 3 * T + 2,
    heat_capacity = lambda T: 4 * T + 8,
    rho = lambda T: 7 * T + 5,
    Q = lambda T: -0.5 * T**2
)
``` 

Concerning MMS test for Soret, this is what is currently done in tests: https://github.com/festim-dev/FESTIM/blob/2ae06c1357194c105d6e4d415c88d105a11258fd/test/system/test_system.py#L466-L555

Aside from that, there is a minor drop in the code coverage. Is it OK?

[the remark this answers is not in the document]
Comment (RemDelaporteMathurin) [reply festim/materials/material.py:21]: > wouldn't it be beneficial to provide a more general example

Yes

> Aside from that, there is a minor drop in the code coverage. Is it OK?

This is due to the removal of some code. This often happens during refactoring. What I'm looking at first is the patch coverage: how many lines that were modified are covered. And the patch coverage is 100% so we're good.

[the remark this answers is not in the document]
Comment (RemDelaporteMathurin) [reply festim/materials/material.py:21]: Yes @KulaginVladimir you are right this is the MMS test we have for the Soret effect.

**Review by RemDelaporteMathurin — APPROVED** [review]
Thanks @KulaginVladimir 🚀  I'll leave this open to leave others the time to make some comments if needed.

**Pull request merged**